### ISSUE TITLE
Tag autocompletion component improvements: UI, UX & a11y

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#14845]
These improvements attempt to solve the following user problems:
- When I type, I don't know if I have added a tag successfully.
- Tag description sometimes gets too long that I can't read other tags easily. 
- Tag autocomplete is not accessible

Links
- [Epic](https://github.com/forem/rfcs/issues/307)
- [forem/forem issue](https://github.com/forem/forem/issues/14845)
- [Figma spec](https://www.figma.com/file/RzV8pYcgujohbdTjYOKqld/Tags?node-id=0%3A1)

## Tasks

- [x] #15421
- [x] #15809
- [x] #15422
- [x] #15423

**Comment by aitchiss** [comment]
Adding here for visibility - we had a pod discussion around tag autocomplete features & accessibility. I raised some concerns around whether some features are going to make the UX too confusing for some assistive tech users, particularly: the input moving inside the selection list when we "edit" a selection, and special key behaviours like <kbd>,</kbd> and <kbd>space</kbd> selecting the current value, and <kbd>backspace</kbd> putting the last tag into edit mode.

On balance, we believe the experience will be _usable_ for all users, although perhaps not optimum, and the "extra" features will provide real value to repeat Forem users. Based on this, we're going to go ahead in this first iteration with all the planned features, and remain open to any user feedback we receive.

The base component will conform to [combobox authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/#combobox) so we can have confidence in the core functionality.
